### PR TITLE
Added location search feature

### PR DESCRIPTION
**Pull request by MikeBrichko**
Adding location search feature

### CODE REVIEW

**Comment by felixlapierre** [comment]
Pls fix merge conflicts before review 😃

**Comment by MikeBrichko** [comment]
Oops! I forgot to push my merge conflict fixes before making the pull request

**Comment by FGRCL** [comment]
91 files is huge are they all necessary?

File: app/.idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -0,0 +1,116 @@
+<component name="ProjectCodeStyleConfiguration">
```
Comment (FGRCL) [inline app/.idea/codeStyles/Project.xml]: I think all the .idea/ files should be added to the gitignore

**Review by FGRCL — CHANGES_REQUESTED** [review]

File: app/build.gradle [hunk 2]
```diff
@@ -30,11 +30,11 @@ android {
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
-    implementation 'androidx.appcompat:appcompat:1.0.2'
-    implementation 'androidx.core:core-ktx:1.0.2'
+    testImplementation 'junit:junit:4.12'
```
Comment (felixlapierre) [inline app/build.gradle:37]: Duplicate of line 41 in this file

File: .idea/codeStyles/Project.xml [hunk 3]
```diff
@@ -1,8 +1,5 @@
 <component name="ProjectCodeStyleConfiguration">
   <code_scheme name="Project" version="173">
-    <AndroidXmlCodeStyleSettings>
```
Comment (felixlapierre) [inline .idea/codeStyles/Project.xml]: Is this necessary?

File: .idea/gradle.xml [hunk 4]
```diff
@@ -1,8 +1,11 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
+  <component name="GradleMigrationSettings" migrationVersion="1" />
```
Comment (felixlapierre) [inline .idea/gradle.xml]: Are these changes necessary?

File: app/build.gradle [hunk 5]
```diff
@@ -30,11 +30,11 @@ android {
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
-    implementation 'androidx.appcompat:appcompat:1.0.2'
-    implementation 'androidx.core:core-ktx:1.0.2'
```
Comment (felixlapierre) [inline app/build.gradle:38]: Are we sure we don't need this?

File: app/local.properties [hunk 6]
```diff
@@ -0,0 +1,8 @@
+## This file must *NOT* be checked into Version Control Systems,
+# as it contains information specific to your local configuration.
+#
+# Location of the SDK. This is only used by Gradle.
+# For customization when using a Version Control System, please read the
+# header note.
+#Mon Jan 20 09:09:31 EST 2020
+sdk.dir=C\:\\Users\\wmike\\AppData\\Local\\Android\\sdk
```
Comment (felixlapierre) [inline app/local.properties]: Definitely shouldn't be here

File: app/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -1,40 +1,32 @@
...
+    <uses-permission android:name="android.permission.INTERNET" />
     <application
         android:allowBackup="true"
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
+
+        <uses-library android:name="org.apache.http.legacy" android:required="false"/>
```
Comment (felixlapierre) [inline app/src/main/AndroidManifest.xml:17]: Is this necessary?

File: app/src/main/java/com/example/campusguide/MapsActivity.kt [hunk 8]
```diff
@@ -1,49 +1,83 @@
...
+import com.google.android.gms.common.ConnectionResult
+import com.google.android.gms.location.LocationListener
+import com.google.android.gms.location.LocationRequest
+import com.google.android.libraries.places.api.Places
+import com.google.android.libraries.places.api.model.Place
+import com.google.android.libraries.places.widget.Autocomplete
+import com.google.android.libraries.places.widget.AutocompleteActivity
+import com.google.android.libraries.places.widget.model.AutocompleteActivityMode
 import com.google.android.material.floatingactionbutton.FloatingActionButton
+import kotlinx.android.synthetic.main.activity_maps.*
```
Comment (felixlapierre) [inline app/src/main/java/com/example/campusguide/MapsActivity.kt:38]: Organize the imports (android studio should have an option to order them)

File: app/src/main/java/com/example/campusguide/MapsActivity.kt [hunk 9]
```diff
@@ -1,49 +1,83 @@
...
+    private var mCurrLocationMarker: Marker? = null
+    private var mGoogleApiClient: GoogleApiClient? = null
+    private lateinit var mLocationRequest: LocationRequest
     private lateinit var fusedLocationClient: FusedLocationProviderClient
     private lateinit var route: Route
 
+    companion object {
+        private const val LOCATION_PERMISSION_ACCESS_CODE = 1
+        private const val AUTOCOMPLETE_REQUEST_CODE = 2
+    }
```
Comment (felixlapierre) [inline app/src/main/java/com/example/campusguide/MapsActivity.kt]: Move this to constants.kt

File: app/src/main/java/com/example/campusguide/MapsActivity.kt [hunk 10]
```diff
@@ -1,49 +1,83 @@
...
+import com.google.android.libraries.places.api.model.Place
+import com.google.android.libraries.places.widget.Autocomplete
+import com.google.android.libraries.places.widget.AutocompleteActivity
+import com.google.android.libraries.places.widget.model.AutocompleteActivityMode
 import com.google.android.material.floatingactionbutton.FloatingActionButton
+import kotlinx.android.synthetic.main.activity_maps.*
 
-class MapsActivity : AppCompatActivity(), OnMapReadyCallback {
+class MapsActivity() : AppCompatActivity(), OnMapReadyCallback, LocationListener,
+    GoogleApiClient.ConnectionCallbacks, GoogleApiClient.OnConnectionFailedListener {
```
Comment (felixlapierre) [inline app/src/main/java/com/example/campusguide/MapsActivity.kt:41]: This feature adds a significant amount of code to the already large MapsActivity, can you extract the functionality into another class?

File: app/src/main/java/com/example/campusguide/MapsActivity.kt [hunk 11]
```diff
@@ -77,36 +114,108 @@ class MapsActivity : AppCompatActivity(), OnMapReadyCallback {
...
+            LocationServices.getFusedLocationProviderClient(this)
+        }
     }
-    
 
+    override fun onConnectionSuspended(i: Int) {
+
+    }
+
+    override fun onLocationChanged(location: Location) {
```
Comment (felixlapierre) [inline app/src/main/java/com/example/campusguide/MapsActivity.kt]: What does this function do?

File: app/src/main/java/com/example/campusguide/SwitchCampus.kt [hunk 12]
```diff
@@ -1,29 +1,35 @@
...
         }
     }
 
-    private fun switchCampus(isAtDowntownCampus: Boolean) {
+    private fun switchCampus(isAtDowntownCampus: Boolean, campusName: TextView) {
         if(isAtDowntownCampus){
             val loyCoord = LatLng(45.458153, -73.640490)
             mMap.animateCamera(CameraUpdateFactory.newLatLngZoom(loyCoord, Constants.ZOOM_STREET_LVL))
-
+            campusName.text = "Loyola Campus"
```
Comment (felixlapierre) [inline app/src/main/java/com/example/campusguide/SwitchCampus.kt]: Move to constants or to string resource

**Review by felixlapierre — CHANGES_REQUESTED** [review]

File: .idea/codeStyles/Project.xml (hunk 3 above)
Comment (MikeBrichko) [reply .idea/codeStyles/Project.xml]: I did not physically change this, I will look into what it means

File: app/build.gradle (hunk 2 above)
Comment (MikeBrichko) [reply app/build.gradle:37]: File will be removed

File: app/build.gradle (hunk 5 above)
Comment (MikeBrichko) [reply app/build.gradle:38]: Features still work, I believe we do not need these dependencies

File: app/local.properties (hunk 6 above)
Comment (MikeBrichko) [reply app/local.properties]: I removed it, thank you for spotting that

File: app/src/main/AndroidManifest.xml (hunk 7 above)
Comment (MikeBrichko) [reply app/src/main/AndroidManifest.xml:17]: The apps behavior does not change if I remove this line, should we remove this line?

File: app/src/main/java/com/example/campusguide/MapsActivity.kt (hunk 8 above)
Comment (MikeBrichko) [reply app/src/main/java/com/example/campusguide/MapsActivity.kt:38]: It has been done, will be visible in next commit

File: app/src/main/java/com/example/campusguide/MapsActivity.kt (hunk 10 above)
Comment (MikeBrichko) [reply app/src/main/java/com/example/campusguide/MapsActivity.kt:41]: I will add it in another class

File: app/src/main/java/com/example/campusguide/MapsActivity.kt (hunk 9 above)
Comment (MikeBrichko) [reply app/src/main/java/com/example/campusguide/MapsActivity.kt]: Sounds good!

File: app/src/main/java/com/example/campusguide/MapsActivity.kt (hunk 11 above)
Comment (MikeBrichko) [reply app/src/main/java/com/example/campusguide/MapsActivity.kt]: It does nothing, I will remove it

File: app/src/main/java/com/example/campusguide/SwitchCampus.kt (hunk 12 above)
Comment (MikeBrichko) [reply app/src/main/java/com/example/campusguide/SwitchCampus.kt]: I'll do that!

File: .idea/codeStyles/Project.xml (hunk 3 above)
Comment (felixlapierre) [reply .idea/codeStyles/Project.xml]: I looked into it and it's just a code style guide thing, not critical

**Review by felixlapierre — APPROVED** [review]

File: app/src/main/java/com/example/campusguide/MapsActivity.kt (hunk 10 above)
Comment (felixlapierre) [reply app/src/main/java/com/example/campusguide/MapsActivity.kt:41]: We decided to move the code later

**Review by FGRCL — APPROVED** [review]

File: app/build.gradle [hunk 13]
```diff
@@ -49,7 +49,12 @@ dependencies {
     androidTestImplementation 'com.android.support.test:runner:1.1.0'
     androidTestImplementation 'com.android.support.test.uiautomator:uiautomator-v18:2.1.1'
     androidTestImplementation 'org.hamcrest:hamcrest-library:1.3'
+<<<<<<< HEAD
```
Comment (felixlapierre) [inline app/build.gradle]: You didn't fix this merge conflict. Next time, make sure the build passes before you commit.

**Review by felixlapierre — CHANGES_REQUESTED** [review]
Fix merge conflict

**Review by felixlapierre — APPROVED** [review]

**Pull request merged**